queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/business_info.py

```diff
@@ -57,7 +57,12 @@ def set_association_type(business: Business, association_type: String) -> Dict:
 def set_legal_name(corp_num: str, business: Business, business_info: Dict):
     """Set the legal_name in the business object."""
     legal_name = business_info.get('legalName', None)
-    business.legal_name = legal_name if legal_name else corp_num[2:] + ' B.C. LTD.'
+    if legal_name:
+        business.legal_name = legal_name
+    else:
+        legal_type = business_info.get('legalType', None)
+        numbered_legal_name_suffix = Business.BUSINESSES[legal_type]['numberedLegalNameSuffix']
```

Comment:
Just to confirm, does this retrieve data? https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/models/business.py#L142 This dict expect enum instead of LegalTypes.BCOMP.value